modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -62,16 +68,19 @@
     private final Function<ClientChannelConfiguration, ClientChannel> chFactory;
 
     /** Client channel holders for each configured address. */
-    private final ClientChannelHolder[] channels;
+    private final AtomicReference<List<ClientChannelHolder>> channels = new AtomicReference<>();
```

Comment:
`AtomicReference` is not needed, we can just use `volatile` since only `get()` and `set()` methods are used below.

Reply:
Fixed.